### PR TITLE
A modular skin made to use Mixxx as a player (A really good one :) )

### PR DESCRIPTION
**Pull request by jclaveau**
Howdy everybody,

I discovered Mixxx a year ago and rapidly realized I preferred to play my music with it than the usual Linux players.

But, you know what? It's not a player! The potmeters are really hard to use with a trackpad and the most important information is hidden/reduced by the controls needed to mix. So I decided to make a skin easy to use with no HID controller and giving as much place as possible to the library and the overviews of tracks. A skin thought to be used in autodj mode.

Here are the rules I followed to make it : 
- The potmeters are replaced by faders
- The controls used to mix are hidden by default but available if you need them
- The library is as big as possible

The controls are available this way : 
- The master volume is always available but expandable to display the prelisten controls (headphone volume and mix) and the balance.
- The tracks vumeters are expandable to make the gains and volumes available.
- The play/pause and sync controls can be expanded to control the pitch, the loops, the cues and associated buttons (quantization, keylock...).
- The killers are always visible and are expanded by the equalizers.
- The preview is hideable (Actually the overviews and the previews are in a splitter so you can choose how much you want to see them).

Now its limits : 
- I use it as a player so there is only two decks.
- Sadly, I hadn't enough time to prepare the samplers and effects views but I will eventually.

Two screenshots, one with all controls hidden and the other fully opened : 
mixxx-expandable-mini

mixxx-expandable-full

As I'm not a designer, I decided to use SVG. It's definitely easier to draw with markup language than a dedicated software! hum... for me at least xD

Learning how to skin Mixxx, I noticed that I needed some features that didn't exist to avoid duplicating the same files dozen times. So I tried to do an svg template engine to manage them. I had to remind my old lessons of C and discover C++ with Qt to do that. So I apologize for all the help/reviews I required to fulfill (and it's still not mergeable!).

You can find this work here https://github.com/mixxxdj/mixxx/pull/302

By the way, there are some big advantages to svg compared to bitmap : 
- The size of this skin is 165Kb.
- I don't have the hardware to test it but it would be fine on a retina screen.
- There is almost one file per type of widget so it's really easy to maintain.

I really think this skin can be useful for people who are not real djs but have found some features they like in this excellent software (Like the case of Seemanta Dutta, the Tango lover :dancer: ). That's why I create this pull request now. An other reason is that it's a living example of how svg can be used to design Mixxx. 

I consider it as a beta as all my needs are served but every feedback would be welcome.

Hoping you'll love it!

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -1,2 +1,3 @@
 *~
 *.orig
+Nikto/
```
Comment (rryan) [inline .gitignore]: Nit -- please remove :). But I'm intrigued as to what's coming next.

**Comment by rryan** [comment]
The file size rocks! SVG will totally shrink our binary package sizes.

**Comment by jclaveau** [comment]
25Kb zipped in 7z :)

**Comment by daschuer** [comment]
I think we can merge it to developer_skins at any time since this branch is an experimental one. 
Any objections? 

I do not know if its suits for merging to master as well once it is finished. 
Maybe @esbrandt can give some comments.

**Comment by daschuer** [comment]
the skin comes up on Ubuntu Trusty like that:
modularubuntu

**Comment by rryan** [comment]
Yea, no objections for merging to developer_skins. The bar should be low or
zero :).

**Comment by jclaveau** [comment]
I'm on Xubuntu 14.04 and it requires the svg parser to display correctly. Btw, I still have a lot of work (filters mainly plus understanding some differences between layouts) so I'm not sure it's the good moment for merging, I just wanted to expose the UX experiment and a demo of the svg parser :)

**Pull request merged**